### PR TITLE
finished first go at GLV integration

### PR DESCRIPTION
**Pull request by davebshow**

### CODE REVIEW

File: goblin/engine.py [hunk 1]
```diff
@@ -43,7 +41,7 @@
     msg = await stream.fetch_data()
     features['threaded_transactions'] = msg.data[0]
 
-    return Engine(url, conn, loop, **features)
+    return Engine(url, conn, loop, force_close=force_close, **features)
```
Comment (davebshow) [inline goblin/engine.py:45]: `force_close` functionality will be removed on next driver update.

File: goblin/engine.py [hunk 2]
```diff
@@ -43,7 +41,7 @@
...
```
Comment (leifurhauks) [inline goblin/engine.py:44]: I think I remember talking about connections and responses having `force_close` before we switched to the queue-based responses and multiplexed connections. What does it mean for an Engine to be force_close? Does it execute one request/response cycle and then self-destruct?

Comment (davebshow) [inline goblin/engine.py:44]: See line note below (above in the conversation tab). I think connections should be closed explicitly by closing the engine. In the case of a driver user, connections should be either closed explicitly with `close` or by using a CM

File: goblin/traversal.py [hunk 3]
```diff
@@ -57,19 +65,18 @@ def graph(self):
...
+            label = self._element_class.__mapping__.label
+            traversal = self._graph.traversal()
+            if self._element_class.__type__ == 'vertex':
+                traversal = traversal.V()
+            if self._element_class.__type__ == 'edge':
+                traversal = traversal.E()
+            traversal = traversal.hasLabel(label)
+        else:
+            traversal = self.graph.traversal()
+        return traversal
```
Comment (davebshow) [inline goblin/traversal.py:79]: This function can be a bit more pithy.

**Comment by leifurhauks** [comment]
Looks good so far!
So if I understand correctly, it should something look like this?

``` python
# if MyVertexClass corresponds to label 'my_vertex',
# the following traversal should be equivalent to g.V().hasLabel('my_vertex').has('foo', 'bar')
resp = await session.traversal(MyVertexClass).has(MyVertexClass.foo, 'bar')

async for item in resp:
    """do stuff"""
```

**Comment by davebshow** [comment]
Just about, you need to a "termination method" like: `all()` or `one` as well as getting the mapping for the property name:

``` python
# if MyVertexClass corresponds to label 'my_vertex',
# the following traversal should be equivalent to g.V().hasLabel('my_vertex').has('foo', 'bar')
resp = await session.traversal(MyVertexClass).has(
    MyVertexClass.__mapping__.foo, 'bar').all()

async for item in resp:
    """do stuff"""
```

**Comment by davebshow** [comment]
You can also create "generic traversals" that update session objects when possible:

``` python
resp = await session.g.V().anyOtherSteps.all()
...
```

But for this to really work I still need to add generic deserialization.

**Comment by davebshow** [comment]
Since we don't have a stable master I will merge and move this work to new branches for Element registry/config etc.

**Pull request merged**